### PR TITLE
Code Review: Slow canvas refresh (#10795)

### PR DESCRIPTION
**Pull request by randomsequence**
Code review for Slow canvas refresh (#10795):

> **Sketch Beta Version 40 (33762) / macOS 10.12**
> 
> Since updating to the latest version this morning, I've noticed a sudden performance drop. Everything seems a little slower, especially the canvas refresh, which is suffering the most.
> 
> I've attached below a screen recording and the .sketch file.
> 
> [Screen Recording (Sketch Beta).m4v.zip](https://github.com/BohemianCoding/Sketch/files/482872/Screen.Recording.Sketch.Beta.m4v.zip)
> 
> [Test.zip](https://github.com/BohemianCoding/Sketch/files/482875/Test.zip)

Connect to BohemianCoding/Sketch#10795.

### CODE REVIEW

**Comment by randomsequence** [comment]
Review notes:

The `BCEdgeInsets` code is mostly copied from `UIKit` (`UIGeometry.h`). Tests are my own.

File: Source/BCRectGeometry.h [hunk 1]
```diff
@@ -21,6 +21,33 @@ typedef NS_ENUM(NSInteger, BCRectComparison) {
   BCRectComparisonDefault                 = BCRectComparisonLeftToRightTopToBottom
 };
 
+/// BCEdgeInsets mirrors UIEdgeInsets, in expecting a `top-left` origin.
+typedef struct BCEdgeInsets {
```
Comment (pieteromvlee) [inline Source/BCRectGeometry.h]: This feels like the equivalent data structure as our existing `BCEdgePaddings`

Comment (randomsequence) [inline Source/BCRectGeometry.h]: Doh!

**Comment by randomsequence** [comment]
Closing this, since we don't need a duplicate of `BCEdgePaddings`.

**Pull request closed without merging**